client/my-sites/domains/domain-management/edit/card/renew-button/index.tsx

```diff
@@ -0,0 +1,87 @@
+import { Button } from '@automattic/components';
+import formatCurrency from '@automattic/format-currency';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { handleRenewNowClick, getRenewalPrice } from 'calypso/lib/purchases';
+import type { Purchase } from 'calypso/lib/purchases/types';
+import type { ProductListItem } from 'calypso/state/products-list/selectors/get-products-list';
+import type { SiteData } from 'calypso/state/ui/selectors/site-data';
+import type { ReactChild } from 'react';
+
+import './style.scss';
+
+type RenewButtonProps = {
+	purchase: Purchase | null;
+	selectedSite: SiteData;
+	subscriptionId: number;
+	compact?: boolean;
+	primary?: boolean;
+	disabled?: boolean;
+	redemptionProduct?: ProductListItem;
+	reactivate?: boolean;
+	customLabel?: ReactChild;
+	tracksProps?: Record< string, string >;
+};
+
+function RenewButton( {
+	compact,
+	primary,
+	selectedSite,
+	subscriptionId,
+	redemptionProduct,
+	reactivate,
+	customLabel,
+	tracksProps,
+	purchase,
+	disabled,
```

Comment:
Nit: can we keep the props and type definition in alphabetical order? Or at least the same order? There are a _lot_ of props here!
```suggestion
	compact?: boolean;
	customLabel?: ReactChild;
	disabled?: boolean;
	primary?: boolean;
	purchase: Purchase | null;
	reactivate?: boolean;
	redemptionProduct?: ProductListItem;
	selectedSite: SiteData;
	subscriptionId: number;
	tracksProps?: Record< string, string >;
};

function RenewButton( {
	compact,
	customLabel,
	disabled,
	primary,
	purchase,
	reactivate,
	redemptionProduct,
	selectedSite,
	subscriptionId,
	tracksProps,
```

Reply:
> can we keep the props and type definition in alphabetical order?

OK, I fixed this by dividing the props into two groups: the essential mandatory ones (`purchase`, `selectedSite`, `subscriptionId`) and optional ones, and sorting each group alphabetically.